app/client/components/homePage/banner/Countdown.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { CSSProperties, useEffect } from "react";
+
+interface CountdownProps {
+  endDate: Date;
+  onCountdownFinish: () => void;
+}
+
+interface CountdownSpanCSS extends CSSProperties {
+  "--value": number;
+}
+
+export const Countdown = ({ endDate, onCountdownFinish }: CountdownProps) => {
+  const calculateTimeLeft = () => {
```

Comment:
I'm unsure about that one, as it does more than calculate time. I would remove onCountdownFinish from here, move this method outside of the component (and return an object entirely of 0's if milisDifference < 0). Then, we can check inside the component itself if the time is 0, and if so, call onCountdownFinish.

That solution is also good, so just do what you think here.

Reply:
Done